File: 4.Clients/WebApi/ClientApp/src/entities/offer-stage.ts
import { StageStatusEnum } from './enums/stage-status.enum';
import { SeniorityEnum } from './enums/seniority.enum';
import {Validators} from "@angular/forms";

export class OfferStage {
    id: number;
    date?: Date;
    feedback: string;
    status: StageStatusEnum;
    userOwnerId?: number;
    userDelegateId?: number;
    processId: number;
    rejectionReason?: string;
    hireDate: Date;
    seniority: SeniorityEnum;
    remunerationOffer: number;
    vacationDays: number;
  firstday: Date;
    bonus: string;
    backgroundCheckDone: boolean;
    backgroundCheckDoneDate: Date;
    preocupationalDone: boolean;
    preocupationalDoneDate: Date;
}

File: 1.Domain/Domain.Model/OfferStage.cs
using Core;
using Domain.Model.Enum;
using System;

namespace Domain.Model
{
    public class OfferStage: Entity<int>
    {
        public StageType Type { get; set; }
        public DateTime? Date { get; set; }
        public string Feedback { get; set; }
        public StageStatus Status { get; set; }
        public int ProcessId { get; set; }
        public Process Process { get; set; }
        public int? UserOwnerId { get; set; }
        public User UserOwner { get; set; }
        public int? UserDelegateId { get; set; }
        public User UserDelegate { get; set; }
        public string RejectionReason { get; set; }
        public DateTime HireDate { get; set; }
        public Seniority Seniority { get; set; }      
        public bool BackgroundCheckDone { get; set; }
        public DateTime? BackgroundCheckDoneDate { get; set; }
        public bool PreocupationalDone { get; set; }
        public DateTime? PreocupationalDoneDate { get; set; }
        public int RemunerationOffer { get; set; }
        public int VacationDays { get; set; }
        public DateTime Firstday { get; set; }
        public string Bonus { get; set; }
        public HealthInsuranceEnum HealthInsurance { get; set; }
        public string Notes { get; set; }
    }
}
